src/fluree/db/ledger/indexing.clj

```diff
@@ -159,75 +168,86 @@
            :first first-flake
            :rhs rhs)))
 
-(defn rebalancer [idx t novelty remove-preds]
+(defn rebalance-children
+  [branch idx child-nodes]
+  (let [target-count (/ *overflow-children* 2)]
+    (->> child-nodes
+         (partition-all target-count)
+         (map (fn [kids]
+                (update-branch branch idx kids)))
+         (fn [[maybe-leftmost & not-leftmost]]
+           (into [maybe-leftmost] (map (fn [non-left-node]
+                                         (assoc non-left-node
+                                                :leftmost? false)))
+                 not-leftmost)))))
+
+(defn integrate-novelty
+  "Returns a transducer that transforms a stream of index nodes in depth first
+  order by incorporating the novelty flakes into the nodes, removing flakes with
+  predicates in remove-preds, rebalancing the leaves so that none is bigger than
+  *overflow-bytes*, and rebalancing the branches so that none have more children
+  than *overflow-children*."
+  [idx t novelty remove-preds]
   (fn [xf]
     (let [stack (volatile! [])]
       (fn
         ;; Initialization: do nothing but initialize the nested transformer by
-        ;; calling it's initializer.
+        ;; calling it's initializing fn.
         ([]
          (xf))
 
-        ;; Completion: Flush the stack, calling the nested transformer's
-        ;; iterator to iterate each remaining item before calling the nested
-        ;; transformer's completion fn on the result of the iterations.
-        ([result]
-         (loop [result* result]
-           (if-let [nxt (peek @stack)]
-             (do (vswap! stack pop)
-                 (recur (unreduced (xf result* nxt))))
-             (xf result*))))
-
         ;; Iteration:
-        ([result nxt]
-         (if (index/leaf? nxt)
-           (let [new-flakes (index/novelty-subrange nxt t novelty)
-                 to-remove  (if (seq remove-preds)
-                              (->> new-flakes
-                                   (concat (:flakes nxt))
-                                   (filter (fn [f]
-                                             (contains? remove-preds
-                                                        (flake/p f)))))
-                              [])]
+        ;;   1. Incorporate each successive node with it's corresponding novelty
```

Comment:
Want to applaud the docs here, they are very helpful. Got an extraneous apostrophe, though :p